implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java

```diff
@@ -93,11 +96,16 @@ public SmallRyeConfigBuilder addDiscoveredInterceptors() {
         return discoveredConverters;
     }
 
-    List<ConfigSourceInterceptor> discoverInterceptors() {
-        List<ConfigSourceInterceptor> interceptors = new ArrayList<>();
+    List<InterceptorWithPriority> discoverInterceptors() {
```

Comment:
If we're bundling our interceptors, I think we should separate `addDefaultInterceptors` from `addDiscoveredInterceptors` so that we can replace the default ones (for example, configuring them differently) while still discovering user interceptors.

Reply:
This is something we should decide. At this time, we are not bundling any of our interceptors by default, you need to register them via the ServiceLoader.

Do you think we should bundle them by default?

Reply:
Yes I think so.  Otherwise it will be very hard for users to have discovered interceptors while also excluding these initial ones.

Reply:
They don't need to exclude them. They are only discovered if you register them via ServiceLoader and right now we are not registering any of the interceptors we added. Right now the user will need to add the ServiceLoader file to they app with any of our interceptors reference to register them. So, even if they call `addDiscoveredInterceptors` none of ours will be added.

BTW, we don't have a `addDefaultInterceptors` for the same reason, but I can add it.